pkg/probeupload/server_handler.go

```diff
@@ -40,18 +42,36 @@ func LogCallback(logger logging.Logger) func(error) {
 	}
 }
 
-// NewProbeServerHandler returns an http.Handler for serving kernel probes. The handler assumes the path of kernel
-// probes is rooted at `/`, i.e., wrap this via `http.StripPrefix` when serving on a sub-path.
-// The errorCallback is invoked for errors that happen during writing the response body, and thus cannot be transmitted
-// to the client via status/headers. It may be nil, in which case errors are simply ignored.
-func NewProbeServerHandler(errorCallback func(error), sources ...ProbeSource) http.Handler {
-	return &handler{
+// NewProbeServerHandler returns a http.Handler for serving kernel probes.
+// It runs in Online mode by default and is meant to be used in Central.
+// The probeServerHandler assumes the path of kernel probes is rooted at `/`, i.e., wrap this via `http.StripPrefix`
+// when serving on a sub-path. The errorCallback is invoked for errors that happen during writing the response body,
+// and thus cannot be transmitted  to the client via status/headers. It may be nil, in which case errors are ignored.
+func NewProbeServerHandler(errorCallback func(error), sources ...ProbeSource) *probeServerHandler {
+	psh := NewSensorProbeServerHandler(errorCallback, sources...)
+	psh.GoOnline() // this hadler is used in Central as well and it must be immediately online for backwards compat.
+	return psh
+}
+
+// NewSensorProbeServerHandler returns the same http.Handler as in NewProbeServerHandler to be used in Sensor.
+// The difference to NewProbeServerHandler is that it starts in offline mode
+func NewSensorProbeServerHandler(errorCallback func(error), sources ...ProbeSource) *probeServerHandler {
```

Comment:
Since this constructor will be called in the central probe handler, maybe let's remove `Sensor` from its name? 
```suggestion
func NewConnectionAwareProbeHandler(errorCallback func(error), sources ...ProbeSource) *probeServerHandler {
```

Reply:
There are two constructors - one with Sensor, other without. I added `Sesor` in the name on purpose to avoid calling it from Central.

Reply:
> I added Sensor in the name on purpose to avoid calling it from Central.

But you are calling the "`Sensor`" constructor in `NewProbeServerHandler` (which the comments says it's called by Central). Hence the confusion.

Reply:
Oh, this way. Yeah, I am calling it to reuse some code only. I could copy the construction code to avoid calling one constructor from the other. Or simply rename it...

Reply:
I applied the rename in [566e8a9](https://github.com/stackrox/stackrox/pull/7349/commits/566e8a9a4d50a429e418467343a451047ae67f7c)